### ISSUE TITLE
Obsolete version of the plug-in protocol

### ISSUE THREAD

**Issue by flyisland** [ryan-robeson/homebrew-gimp#2]
I've successfully installed the plugin by following commands:

```
brew uninstall resynthesizer
brew uninstall libgimp2.0
brew install --build-from-source libgimp2.0
brew install --build-from-source resynthesizer
```

but got below error while trying to apply this plugin
![Screen Shot 2020-01-09 at 10 15 57](https://user-images.githubusercontent.com/547085/72032133-7cb0d380-32c9-11ea-8c7a-33548940b3b7.png)

I'm using the "GIMP 2.10.14" on "macOS mojave version 0.14.6 (18G2022)".

**Comment by ryan-robeson** [comment]
Thanks for sticking with me on this.

I'm guessing that error may be caused by GIMP being compiled with newer versions of some libraries than Homebrew provides for us to compile libgimp and resynthesizer with.

To check, could you post the output of:
```
otool -L $(brew --prefix)/share/gimp-plugins/resynthesizer/resynthesizer)

otool -L /Applications/GIMP-2.10.app/Contents/MacOS/gimp-bin 
```

`otool -L` displays the "names and version numbers of  the  shared  libraries  that  the  object file uses, as well as the shared library ID if the file is a shared library", so that should show if GIMP is using newer libraries.

**Comment by ryan-robeson** [comment]
Correction:
The second command may just be 
`otool -L /Applications/GIMP-2.10.app/Contents/MacOS/gimp`

Without the '-bin'

**Comment by flyisland** [comment]
```
bash-3.2$ otool -L $(brew --prefix)/share/gimp-plugins/resynthesizer/resynthesizer
/usr/local/share/gimp-plugins/resynthesizer/resynthesizer:
	/usr/local/opt/libgimp2.0/lib/libgimpui-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.10.0)
	/usr/local/opt/libgimp2.0/lib/libgimpwidgets-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.10.0)
	/usr/local/opt/libgimp2.0/lib/libgimpmodule-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.10.0)
	/usr/local/opt/libgimp2.0/lib/libgimp-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.10.0)
	/usr/local/opt/libgimp2.0/lib/libgimpmath-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.10.0)
	/usr/local/opt/libgimp2.0/lib/libgimpconfig-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.10.0)
	/usr/local/opt/libgimp2.0/lib/libgimpcolor-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.10.0)
	/usr/local/opt/libgimp2.0/lib/libgimpbase-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.10.0)
	/usr/local/opt/gegl/lib/libgegl-0.4.0.dylib (compatibility version 417.0.0, current version 417.0.0)
	/usr/local/opt/gegl/lib/libgegl-npd-0.4.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.250.1)
	/usr/local/opt/glib/lib/libgmodule-2.0.0.dylib (compatibility version 6201.0.0, current version 6201.4.0)
	/usr/local/opt/json-glib/lib/libjson-glib-1.0.0.dylib (compatibility version 401.0.0, current version 401.4.0)
	/usr/local/opt/babl/lib/libbabl-0.1.0.dylib (compatibility version 168.0.0, current version 168.1.0)
	/usr/local/opt/gtk+/lib/libgtk-quartz-2.0.0.dylib (compatibility version 2401.0.0, current version 2401.32.0)
	/usr/local/opt/gtk+/lib/libgdk-quartz-2.0.0.dylib (compatibility version 2401.0.0, current version 2401.32.0)
	/System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa (compatibility version 1.0.0, current version 23.0.0)
	/usr/local/opt/pango/lib/libpangocairo-1.0.0.dylib (compatibility version 4401.0.0, current version 4401.7.0)
	/usr/local/opt/pango/lib/libpango-1.0.0.dylib (compatibility version 4401.0.0, current version 4401.7.0)
	/usr/local/opt/harfbuzz/lib/libharfbuzz.0.dylib (compatibility version 20601.0.0, current version 20601.4.0)
	/usr/local/opt/atk/lib/libatk-1.0.0.dylib (compatibility version 23411.0.0, current version 23411.1.0)
	/usr/local/opt/cairo/lib/libcairo.2.dylib (compatibility version 11603.0.0, current version 11603.0.0)
	/usr/local/opt/gdk-pixbuf/lib/libgdk_pixbuf-2.0.0.dylib (compatibility version 4001.0.0, current version 4001.0.0)
	/usr/local/opt/glib/lib/libgio-2.0.0.dylib (compatibility version 6201.0.0, current version 6201.4.0)
	/usr/local/opt/glib/lib/libgobject-2.0.0.dylib (compatibility version 6201.0.0, current version 6201.4.0)
	/usr/local/opt/glib/lib/libglib-2.0.0.dylib (compatibility version 6201.0.0, current version 6201.4.0)
	/usr/local/opt/gettext/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.6.0)
	/usr/local/opt/glib/lib/libgthread-2.0.0.dylib (compatibility version 6201.0.0, current version 6201.4.0)
bash-3.2$ otool -L /Applications/GIMP-2.10.app/Contents/MacOS/gimp
/Applications/GIMP-2.10.app/Contents/MacOS/gimp:
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 853.0.0)
	@rpath/libgimpwidgets-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.14.0)
	@rpath/libgtkmacintegration-gtk2.2.dylib (compatibility version 4.0.0, current version 4.1.0)
	@rpath/libgtk-quartz-2.0.0.dylib (compatibility version 2401.0.0, current version 2401.32.0)
	@rpath/libgdk-quartz-2.0.0.dylib (compatibility version 2401.0.0, current version 2401.32.0)
	@rpath/libatk-1.0.0.dylib (compatibility version 23210.0.0, current version 23210.1.0)
	@rpath/libgimpconfig-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.14.0)
	@rpath/libgimpmath-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.14.0)
	@rpath/libgimpthumb-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.14.0)
	@rpath/libgimpcolor-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.14.0)
	@rpath/libgimpmodule-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.14.0)
	@rpath/libgimpbase-2.0.0.dylib (compatibility version 1001.0.0, current version 1001.14.0)
	@rpath/libgdk_pixbuf-2.0.0.dylib (compatibility version 3801.0.0, current version 3801.1.0)
	@rpath/libpangocairo-1.0.0.dylib (compatibility version 4301.0.0, current version 4301.0.0)
	@rpath/libpangoft2-1.0.0.dylib (compatibility version 4301.0.0, current version 4301.0.0)
	@rpath/libpango-1.0.0.dylib (compatibility version 4301.0.0, current version 4301.0.0)
	@rpath/libfontconfig.1.dylib (compatibility version 14.0.0, current version 14.0.0)
	@rpath/libfreetype.6.dylib (compatibility version 24.0.0, current version 24.1.0)
	@rpath/libharfbuzz.0.dylib (compatibility version 20504.0.0, current version 20504.0.0)
	@rpath/libcairo.2.dylib (compatibility version 11603.0.0, current version 11603.0.0)
	@rpath/libgegl-0.4.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libgegl-npd-0.4.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libgmodule-2.0.0.dylib (compatibility version 6001.0.0, current version 6001.6.0)
	@rpath/libjson-glib-1.0.0.dylib (compatibility version 201.0.0, current version 201.8.0)
	@rpath/libgio-2.0.0.dylib (compatibility version 6001.0.0, current version 6001.6.0)
	@rpath/libbabl-0.1.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/liblcms2.2.dylib (compatibility version 3.0.0, current version 3.8.0)
	@rpath/libgexiv2.2.dylib (compatibility version 3.0.0, current version 3.0.0)
	@rpath/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	@rpath/libmypaint-1.3.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libjson-c.4.dylib (compatibility version 5.0.0, current version 5.0.0)
	@rpath/libgobject-2.0.0.dylib (compatibility version 6001.0.0, current version 6001.6.0)
	@rpath/libglib-2.0.0.dylib (compatibility version 6001.0.0, current version 6001.6.0)
	@rpath/libintl.9.dylib (compatibility version 11.0.0, current version 11.4.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
	@rpath/libjpeg.9.dylib (compatibility version 13.0.0, current version 13.0.0)
	/System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa (compatibility version 1.0.0, current version 20.0.0)
	/System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 1249.0.0)
	/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 600.0.0)
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1053.0.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
```

**Comment by ryan-robeson** [comment]
For reference:

| lib | r_compatible | g_compatible | r_current | g_current |
| --- | ------------ | ------------ | --------- | --------- |
| libatk-1.0.0.dylib | 23411.0.0 | 23210.0.0 | 23411.1.0 | 23210.1.0 |
| libbabl-0.1.0.dylib | 168.0.0 | 0.0.0 | 168.1.0 | 0.0.0 |
| libgdk_pixbuf-2.0.0.dylib | 4001.0.0 | 3801.0.0 | 4001.0.0 | 3801.1.0 |
| libgegl-0.4.0.dylib | 417.0.0 | 0.0.0 | 417.0.0 | 0.0.0 |
| libgio-2.0.0.dylib | 6201.0.0 | 6001.0.0 | 6201.4.0 | 6001.6.0 |
| libglib-2.0.0.dylib | 6201.0.0 | 6001.0.0 | 6201.4.0 | 6001.6.0 |
| libgmodule-2.0.0.dylib | 6201.0.0 | 6001.0.0 | 6201.4.0 | 6001.6.0 |
| libgobject-2.0.0.dylib | 6201.0.0 | 6001.0.0 | 6201.4.0 | 6001.6.0 |
| libharfbuzz.0.dylib | 20601.0.0 | 20504.0.0 | 20601.4.0 | 20504.0.0 |
| libintl.8.dylib | 10.0.0 | 11.0.0 | 10.6.0 | 11.4.0 |
| libjson-glib-1.0.0.dylib | 401.0.0 | 201.0.0 | 401.4.0 | 201.8.0 |
| libpango-1.0.0.dylib | 4401.0.0 | 4301.0.0 | 4401.7.0 | 4301.0.0 |
| libpangocairo-1.0.0.dylib | 4401.0.0 | 4301.0.0 | 4401.7.0 | 4301.0.0 |

_\* GIMP actually references libintl.9 rather than libintl.8_

It looks like Homebrew's libraries are actually newer than the one's used to compile the official distribution of GIMP (I assume you're using the official download anyway).

However, after looking into GIMP's sources a bit, I now believe the problem is as simple as needing to compile resynthesizer against a 2.10.14 based libgimp2.0. Currently libgimp2.0 is compiled against 2.10.10. At some point after 2.10.12, the GIMP protocol version was bumped (from 25 to 26), hence the obsolete message. 

I updated the formula locally to compile against 2.10.14, but babl and gegl have not been updated in Homebrew yet (see homebrew/homebrew-core#45841), so the compile fails.

Once those get updated, I'll try compiling again and see where we are at.

In the meantime, downgrading to GIMP 2.10.12 is the most straightforward way to get resynthesizer running.

**Comment by flyisland** [comment]
thanks, it works now after downgrading to GIMP 2.10.12!

**Comment by ryan-robeson** [comment]
Just to followup on this,

With the recent release of GEGL finally making it into homebrew, I was able to get 2.10.14 and 2.10.18 compiling. In my limited testing, resynthesizer compiled against 2.10.18 works in both 2.10.12 and 2.10.14 versions of GIMP (an official version of 2.10.18 is not yet available on macOS).

I did run into an issue installing the latest version of libgimp2.0 where it wouldn't install from the bottle. Running `brew install --force-bottle libgimp2.0` solved this for me. I'm hoping nobody else is affected by that.
